File: Korepetynder.Contracts/Requests/Users/UserRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Korepetynder.Contracts.Requests.Users
{
    public class UserRequest
    {

        [MaxLength(50)]
        public string FirstName { get; set; }
        [MaxLength(50)]
        public string LastName { get; set; }
        [MaxLength(100)]
        public string Email { get; set; }
        [MaxLength(15)]
        public string? TelephoneNumber { get; set; }
        public int Age { get; set; }
        public UserRequest(string firstName, string lastName, string email, int age, string? telephoneNumber)
        {
            FirstName = firstName;
            LastName = lastName;
            Email = email;
            TelephoneNumber = telephoneNumber;
            Age = age;
        }
    }
}

File: korepetynder-spa/src/app/settings/models/requests/userRequest.ts
import { DateTime } from 'luxon';

export class UserRequest {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  birthDate: string;

  constructor(firstName: string, lastName: string, email: string, phoneNumber: string, birthDate: DateTime) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
    this.phoneNumber = phoneNumber;
    this.birthDate = birthDate.toISO();
  }
}
